x/uibc/quota/keeper/quota.go

```diff
@@ -167,22 +169,36 @@ func (k Keeper) CheckAndUpdateQuota(ctx sdk.Context, denom string, newOutflow sd
 
 func (k Keeper) getExchangePrice(ctx sdk.Context, denom string, amount sdkmath.Int) (sdk.Dec, error) {
 	transferCoin := sdk.NewCoin(denom, amount)
-	var err error
+	var (
+		err          error
+		exchangeRate sdk.Dec
+	)
 
 	// convert to base asset if it is `uToken`
 	if ltypes.HasUTokenPrefix(denom) {
 		transferCoin, err = k.leverageKeeper.ExchangeUToken(ctx, transferCoin)
 		if err != nil {
 			return sdk.Dec{}, err
 		}
-	} else {
-		if _, err := k.leverageKeeper.GetTokenSettings(ctx, denom); err != nil {
-			return sdk.Dec{}, err
-		}
+		//TODO: needs to confirm should i get exchange rate for uToken or not ?
+		// get the exchange rate for uToken
+		// exchangeRate = k.leverageKeeper.DeriveExchangeRate(ctx, transferCoin.Denom)
+	}
+
+	ts, err := k.leverageKeeper.GetTokenSettings(ctx, transferCoin.Denom)
+	if err != nil {
+		return sdk.Dec{}, err
 	}
+	transferCoin = sdk.NewCoin(ts.SymbolDenom, amount)
```

Comment:
why do you overwrite the amout here? You will lose the information you have when you convert uToken -> Token